modules/charon-core/src/test/java/org/wso2/charon3/core/utils/ResourceManagerUtilTest.java

```diff
@@ -0,0 +1,286 @@
+    public void testProcessStartIndex(String startIndexStr, int expectedStartIndex) throws BadRequestException {
+
+        int startIndex = ResourceManagerUtil.processStartIndex(startIndexStr);
+        Assert.assertEquals(startIndex, expectedStartIndex);
+    }
+
+    @DataProvider(name = "dataForAllSimpleMultiValuedAttributes")
+    public Object[][] dataToAllSimpleMultiValuedAttributes() {
+
+        List<String> schemasList = new ArrayList<>();
+        schemasList.add("urn:ietf:params:scim:schemas:core:2.0:User");
+        schemasList.add("urn:ietf:params:scim:schemas:extension:enterprise:2.0:User");
+        AttributeSchema subSubAttributeSchema =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:addresses.city",
+                        "city", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subSubAttributeSchema1 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1:attribute2.value",
+                        "value", STRING, true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        ArrayList<AttributeSchema> subSubAttributeSchemaList = new ArrayList<>();
+        subSubAttributeSchemaList.add(subSubAttributeSchema);
+        ArrayList<AttributeSchema> subSubAttributeSchemaList1 = new ArrayList<>();
+        subSubAttributeSchemaList.add(subSubAttributeSchema1);
+        AttributeSchema subAttributeSchema1 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:addresses",
+                        "addresses", COMPLEX, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null,
+                        subSubAttributeSchemaList);
+        AttributeSchema subAttributeSchema2 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User:department",
+                        "department", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subAttributeSchema3 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:attribute3",
+                        "attribute3", STRING, true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema subAttributeSchema4 =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1:attribute2",
+                        "attribute2", STRING, false, "", false, false,
```

Comment:
```suggestion
                        "subattribute", STRING, false, "", false, false,
```

Reply:
fixed in 9c935eb718ab31db04618fe9e0880afe22478953